ksqldb-tools/src/test/java/io/confluent/ksql/tools/migrations/MigrationsTest.java

```diff
@@ -276,6 +312,37 @@ private static void initializeAndVerifyMetadataStreamAndTable(final String confi
     ));
   }
 
+  private static void cleanAndVerify(final String configFile) {
+    // Given:
+    assertThat(sourceExists(MIGRATIONS_STREAM, false), is(true));
+    assertThat(sourceExists(MIGRATIONS_TABLE, true), is(true));
+
+    // When: use `clean` to clean up metadata stream and table
+    final int status = MIGRATIONS_CLI.parse("--config-file", configFile, "clean").run();
+    assertThat(status, is(0));
+
+    // Then:
+    assertThatEventually(() -> sourceExists(MIGRATIONS_STREAM, false), is(false));
+    assertThat(sourceExists(MIGRATIONS_TABLE, true), is(false));
+  }
+
+  private static boolean sourceExists(final String sourceName, final boolean isTable) {
+    final String sourceType = isTable ? "TABLE" : "STREAM";
+    final List<KsqlEntity> entities = makeKsqlRequest("LIST " + sourceType + "S;");
+    assertThat(entities, hasSize(1));
```

Comment:
I'm having trouble figuring out why this works - shouldn't the streams FOO and BAR also be listed here?

Reply:
There's only one entity returned from the ksql server. That one entity is a list of streams (or tables) which may contain multiple streams (or tables).